apheleia.el

```diff
@@ -291,26 +297,38 @@ succeeds provided that its exit status is 0."
                  :sentinel
                  (lambda (proc _event)
                    (unless (process-live-p proc)
-                     (with-current-buffer stderr
-                       (when (= 0 (buffer-size))
-                         (insert "[No output received on stderr]\n")))
-                     (unwind-protect
-                         (if (funcall
-                              (or exit-status
-                                  (lambda (status)
-                                    (= 0 status)))
-                              (process-exit-status proc))
-                             (when callback
-                               (funcall callback stdout))
-                           (message
-                            (concat
-                             "Failed to run %s: exit status %s "
-                             "(see hidden buffer%s)")
-                            (car command)
-                            (process-exit-status proc)
-                            stderr))
-                       (when ensure
-                         (funcall ensure)))))))
+                     (let ((exit-ok (funcall
+                                     (or exit-status
+                                         (lambda (status)
+                                           (= 0 status)))
+                                     (process-exit-status proc))))
+                       ;; Append standard-error from current formatter
+                       ;; to log buffer when `apheleia-verbose' or the
+                       ;; formatter failed. Every process output is
+                       ;; delimeted by a line-feed character.
+                       (with-current-buffer log
+                         (when (or apheleia-verbose
+                                   (not exit-ok))
+                           (if (= 0 (buffer-size))
+                               (insert "[No output received on stderr]")
+                             (insert-buffer-substring stderr))
+                           (insert "\n\C-l\n")))
+
+                       (unwind-protect
+                           (if exit-ok
+                               (when callback
+                                 (funcall callback stdout))
+                             (message
+                              (concat
+                               "Failed to run %s: exit status %s "
+                               "(see hidden buffer%s)")
+                              (car command)
+                              (process-exit-status proc)
+                              log))
+                         (when ensure
+                           (funcall ensure))
+                         (kill-buffer stdout)
```

Comment:
Only thing that might be worrying is we kill the stdout buffer but callback might pass it as stdin to a new formatter process. I tried testing it out with python (using isort and black) with a really large python file and got no error so I don't think it should be an issue but it might be worth investigating.